samples/javascript_nodejs/02.echobot-with-counter/bot.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes } = require('botbuilder');
+
+// Turn counter property
+const TURN_COUNTER_PROPERTY = 'turnCounterProperty';
+
+class Bot {
+    /**
+     *
+     * @param {ConversationState} conversation state object
+     */
+    constructor(conversationState) {
+        // creates a new state accessor property.see https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors
+        this.countProperty = conversationState.createProperty(TURN_COUNTER_PROPERTY);
+        this.conversationState = conversationState;
+    }
+    /**
+     *
+     * @param {TurnContext} on turn context object.
+     */
+    async onTurn(turnContext) {
+        // see https://aka.ms/about-bot-activity-message to learn more about the message and other activity types
+        if (turnContext.activity.type === ActivityTypes.Message) {
+            // read from state.
+            let count = await this.countProperty.get(turnContext);
+            count = count === undefined ? 1 : ++count;
+            await turnContext.sendActivity(`${ count }: You said "${ turnContext.activity.text }"`);
+            // increment and set turn counter.
+            await this.countProperty.set(turnContext, count);
+        } else {
+            await turnContext.sendActivity(`[${ turnContext.activity.type } event detected]`);
+        }
+        // Save state changes
+        await this.conversationState.saveChanges(turnContext);
+    }
+}
+
+module.exports.Bot = Bot;
```

Comment:
Shorthand for this can be:

```js
exports.Bot = Bot;
```

Reply:
Per StevenIc, we want to use the named exports pattern.  So:
module.exports.Bot = Bot;  

This forces casing for the associated require();
const { Bot } = require('./bot');